geode-redis/src/integrationTest/java/org/apache/geode/redis/general/LoopingThreads.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.redis.general;
+
+import java.util.Arrays;
+import java.util.concurrent.CountDownLatch;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+public class LoopingThreads {
+  private final int iterationCount;
+  private final Function<Integer, Object>[] functions;
+
+  @SuppressWarnings("unchecked")
+  public LoopingThreads(int iterationCount,
+      Function... functions) {
+    this.iterationCount = iterationCount;
+    this.functions = (Function<Integer, Object>[]) functions;
+  }
+
+  public void run() {
+    CountDownLatch latch = new CountDownLatch(1);
+    Stream<LoopingThread> loopingThreadStream = Arrays
+        .stream(functions)
+        .map((r) -> new LoopingThread(r, iterationCount, latch))
+        .map((t) -> {
+          t.start();
+          return t;
+        });
+
+    latch.countDown();
+
+    loopingThreadStream.forEach(loopingThread -> {
+      try {
+        loopingThread.join();
+      } catch (InterruptedException e) {
+        throw new RuntimeException(e);
+      }
+    });
+
+  }
+
+  private class LoopingRunnable implements Runnable {
+    private final Function<Integer, Object> runnable;
+    private final int iterationCount;
+    private CountDownLatch startLatch;
+
+    public LoopingRunnable(Function<Integer, Object> runnable, int iterationCount,
+        CountDownLatch startLatch) {
+      this.runnable = runnable;
+      this.iterationCount = iterationCount;
+      this.startLatch = startLatch;
+    }
+
+    @Override
+    public void run() {
+      try {
+        startLatch.await();
+      } catch (InterruptedException e) {
+        throw new RuntimeException(e);
+      }
+      for (int i = 0; i < iterationCount; i++) {
+        runnable.apply(i);
+        Thread.yield();
```

Comment:
Should the call of Thread.yield be optional? I'm not sure you would always want this

Reply:
That makes sense. I want to leave it like this as the default, though, and if someone wants to change the behavior they can add that functionality.  How do you feel about that?  Or do you think not having this as an option is a dealbreaker.

Reply:
not a deal breaker. I just merged in a new dunit test for concurrent testing of SADD. It would be nice to have this encapsulated concurrency code in that test before it becomes the pattern for subsequent tests

Reply:
alright.  it's a real PR now.